File: Backend/AICareerHub.API/AICareerHub.API/Models/JobApplication.cs
namespace AICareerHub.API.Models
{
    public class JobApplication
    {
        public Guid Id { get; set; }

        public Guid UserId { get; set; }

        public string CompanyName { get; set; } = string.Empty;

        public string JobTitle { get; set; } = string.Empty;

        public string? JobUrl { get; set; }

        public string Location { get; set; } = string.Empty;

        public string Status { get; set; } = string.Empty;

        public DateOnly AppliedDate { get; set; }

        public string? Notes { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }

        public User User { get; set; } = null!;
    }
}

File: Frontend/aicareerhub-ui/src/app/core/models/job-application.ts
export interface JobApplication {
  id: string;
  companyName: string;
  jobTitle: string;
  jobUrl?: string | null;
  location: string;
  status: string;
  appliedDate: string;
  notes?: string | null;
  createdAt: string;
  updatedAt: string;
}